packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -687,69 +689,160 @@ export function getPathOverrideForPackageManager({
    */
   path: string | undefined;
 } {
-  const no_override = {
-    detectedLockfile: undefined,
-    detectedPackageManager: undefined,
-    path: undefined,
+  const detectedPackageManger = detectPackageManager(
+    cliType,
+    lockfileVersion,
+    nodeVersion
+  );
+  if (!detectedPackageManger) {
+    return NO_OVERRIDE;
+  }
+
+  if (!corepackPackageManager) {
+    return detectedPackageManger;
+  }
+
+  if (
+    validateVersionOverlap(
+      detectedPackageManger.detectedPackageManager,
+      corepackPackageManager
+    )
+  ) {
+    // corepack is going to take care of it; do nothing special
+    return NO_OVERRIDE;
+  }
+
+  throw new Error(
+    `Detected package manager (by lockfile) "${detectedPackageManger.detectedLockfile}" does not match intended corepack package manager "${corepackPackageManager}". Update your lockfile or "package.json#packageManager" values to match.`
+  );
+}
```

Comment:
Do we have a test case for this throw?

Reply:
Tests for this throw were [added here](https://github.com/vercel/vercel/blob/2a4e1f3d48ed0e8b750204f5fc5d35c76caf9bc6/packages/build-utils/test/unit.get-env-for-package-manager.test.ts#L486).